07-routing-gossip.md

```diff
@@ -111,90 +118,99 @@ announcement message; this is accomplished by having a signature from each
+    - MUST NOT add the channel to its local network view.
+    - SHOULD NOT forward the announcement.
+  - if the `short_channel_id`'s output does NOT correspond to a P2WSH (using
+    `bitcoin_key_1` and `bitcoin_key_2`, as specified in
+    [BOLT #3](03-transactions.md#funding-transaction-output)) OR the output is
+    spent:
+    - MUST ignore the message.
+  - if the specified `chain_hash` is unknown to the receiver:
+    - MUST ignore the message.
+  - otherwise:
+    - if `bitcoin_signature_1`, `bitcoin_signature_2`, `node_signature_1` OR
+    `node_signature_2` are invalid OR NOT correct:
+      - SHOULD fail the connection.
+    - otherwise:
+      - if `node_id_1` OR `node_id_2` are blacklisted:
+        - SHOULD ignore the message.
+      - otherwise:
+        - if the transaction referred to was NOT previously announced as a
+        channel:
+          - SHOULD queue the message for rebroadcasting.
+          - MAY choose NOT to for messages longer than the minimum expected
+          length.
+      - if it has previously received a valid `channel_announcement`, for the
+      same transaction, in the same block, but for a different `node_id_1` or
+      `node_id_2`:
+        - SHOULD blacklist the previous message's `node_id_1` and `node_id_2`,
+        as well as this `node_id_1` and `node_id_2` AND forget any channels
+        connected to them.
+      - otherwise:
+        - SHOULD store this `channel_announcement`.
+  - once its funding output has been spent OR reorganized out:
+    - SHOULD forget a channel.
 
 ### Rationale
 
-Requiring both nodes to sign indicates they are both willing to route
-other payments via this channel (i.e. be part of the public network).
-Requiring the Bitcoin signatures proves that they control the channel.
+Both nodes are required to sign to indicate they are willing to route other
+payments via this channel (i.e. be part of the public network); requiring their
+Bitcoin signatures proves that they control the channel.
 
-The blacklisting of conflicting nodes disallows multiple
-different announcements: no node should ever do this, as it implies
-that keys have leaked.
+The blacklisting of conflicting nodes disallows multiple different
+announcements. Such conflicting announcements should never be broadcast by any
+node, as this implies that keys have leaked.
 
-While channels shouldn't be advertised before they are sufficiently
-deep, the requirement against rebroadcasting only applies if the
-transaction hasn't moved to a different block.
+While channels should not be advertised before they are sufficiently deep, the
+requirement against rebroadcasting only applies if the transaction has not moved
+to a different block.
 
-To avoid having to store excessive-sized messages, yet allow
-reasonable expansion in future, nodes are allowed to restrict
-rebroadcasting (perhaps statistically).
+In order to avoid storing excessively-large messages, yet still allow for
```

Comment:
"excessively large" (no hyphens for compounds with adverbs)

Reply:
fixed